packages/form-state-manager/src/tests/utils/use-subscription.test.js

```diff
@@ -211,4 +211,77 @@ describe('useSubscription', () => {
       expect(renderCount).toEqual(2);
     });
   });
+
+  describe('clearedValue', () => {
+    let CleanButton;
+
+    beforeEach(() => {
+      CleanButton = (props) => {
+        const [, onChange] = useSubscription(props);
+
+        return <button onClick={() => onChange(undefined)}>clear</button>;
+      };
+    });
+
+    it('should clearValue on field level', async () => {
+      const managerApi = createManagerApi({});
+
+      const wrapper = mount(
+        <FormManagerContext.Provider value={{ ...managerApi(), formOptions: managerApi }}>
+          <CleanButton name="field" value="some value" clearedValue={null} />
+        </FormManagerContext.Provider>
+      );
+
+      await act(async () => {
+        wrapper.find('button').simulate('click');
+      });
+      wrapper.update();
+
+      expect(managerApi().values.field).toEqual(null);
+    });
+
+    it('should clearValue on form level', async () => {
+      const managerApi = createManagerApi({});
+
+      const wrapper = mount(
+        <FormManagerContext.Provider value={{ ...managerApi(), formOptions: managerApi, clearedValue: null }}>
+          <CleanButton name="field" value="some value" />
+        </FormManagerContext.Provider>
+      );
+
+      await act(async () => {
+        wrapper.find('button').simulate('click');
+      });
+      wrapper.update();
+
+      expect(managerApi().values.field).toEqual(null);
+    });
+
+    it('field cleared value has higher priority', async () => {
+      const managerApi = createManagerApi({});
+
+      const wrapper = mount(
+        <FormManagerContext.Provider value={{ ...managerApi(), formOptions: managerApi, clearedValue: null }}>
+          <CleanButton name="field" value="some value" clearedValue="cleared" />
+        </FormManagerContext.Provider>
+      );
+
+      await act(async () => {
+        wrapper.find('button').simulate('click');
+      });
+      wrapper.update();
+
+      expect(managerApi().values.field).toEqual('cleared');
+    });
+  });
+
+  it('#checkEmpty - edge cases', () => {
+    expect(checkEmpty(false)).toEqual(false);
+    expect(checkEmpty('A')).toEqual(false);
+    expect(checkEmpty(0)).toEqual(false);
+
+    expect(checkEmpty([])).toEqual(true);
+    expect(checkEmpty('')).toEqual(true);
+    expect(checkEmpty(undefined)).toEqual(true);
```

Comment:
So the big question here is. How do we treat null and empty objects?

Reply:
It's the same as before, just copy pasted code with a new test to get it covered. :man_shrugging: :smile: If we don't want to introduce breaking changes, the question is then already answered. :smile: